sdk/resources/mgmt/src/main/java/com/azure/resourcemanager/resources/fluentcore/utils/Utils.java

```diff
@@ -115,20 +123,48 @@ public static String createOdataFilterForTags(String tagName, String tagValue) {
      * Download a file asynchronously.
      *
      * @param url the URL pointing to the file
-     * @param retrofit the retrofit client
+     * @param httpPipeline the http pipeline
      * @return an Observable pointing to the content of the file
      */
+    public static Mono<byte[]> downloadFileAsync(String url, HttpPipeline httpPipeline) {
+        FileService service = RestProxy.create(FileService.class, httpPipeline);
+        try {
+            return service.download(getHost(url), getPathAndQuery(url))
+                .flatMap(response -> FluxUtil.collectBytesInByteBufferStream(response.getValue()));
+        } catch (MalformedURLException ex) {
+            return Mono.empty();
+        }
+    }
+
     /**
-     * Download a file asynchronously.
+     * Get host from url.
      *
-     * @param url the URL pointing to the file
-     * @param retrofit the retrofit client
-     * @return an Observable pointing to the content of the file
+     * @param urlString the url string
+     * @return the host
+     * @throws MalformedURLException when url is invalid format
      */
-    public static Mono<byte[]> downloadFileAsync(String url, HttpPipeline retrofit) {
-        FileService service = RestProxy.create(FileService.class, retrofit);
-        Mono<HttpResponse> response = service.download(url);
-        return response.flatMap(httpResponse -> httpResponse.getBodyAsByteArray());
+    public static String getHost(String urlString) throws MalformedURLException {
+        URL url = new URL(urlString);
+        String protocol = url.getProtocol();
+        String host = url.getAuthority();
+        return protocol + "://" + host;
+    }
+
+    /**
+     * Get path from url.
+     *
+     * @param urlString the url string
+     * @return the path
+     * @throws MalformedURLException when the url is invalid format
+     */
+    public static String getPathAndQuery(String urlString) throws MalformedURLException {
+        URL url = new URL(urlString);
+        String path = url.getPath();
+        String query = url.getQuery();
```

Comment:
I think it should be `getRawPath` and `getRawQuery` due to you just do a string add.

Reply:
Not available for URL. Such methods are in URI.

Reply:
Could use a bit testing (send one to httpbin.org).